draft-ietf-jsonpath-base.md

```diff
@@ -689,7 +687,7 @@ objects, only arrays, or both.
 The resulting nodelist of a child selector is the concatenation of
 the nodelists from each of its //PICKERS// in the order that the //PICKERS//
 appear in the list.
```

Comment:
> The resulting nodelist of a child selector with a single //PICKER// applied to a value is the collection of children of the value matched by the //PICKER//.

I don't think it's necessary to call out operation of the selector with a single child.

> If the value is an array, the resultant nodelist is in array order. If the value is an object, the order of the resultant nodelist is not stipulated, since JSON objects are unordered.

This is addressed in the *pickers that can select multiple values (slice & filter).  I don't think it makes sense outside of that context.  The operation here isn't precisely value selection; it's concatenating the results of the *pickers.

Reply:
> > The resulting nodelist of a child selector with a single //PICKER// applied to a value is the collection of children of the value matched by the //PICKER//.
> 
> I don't think it's necessary to call out operation of the selector with a single *picker.

I don't see how the behaviour is well defined if we leave this out. It seems to me that the spec is silent on this matter.

> 
> > If the value is an array, the resultant nodelist is in array order. If the value is an object, the order of the resultant nodelist is not stipulated, since JSON objects are unordered.
> 
> This is addressed in the *pickers that can select multiple values (slice & filter). I don't think it makes sense outside of that context. The operation here isn't precisely value selection; it's concatenating the results of the *pickers.

I checked the text of the filter *picker and it doesn't specify the result order. It talks about iterating, but doesn't tie down the order of iteration. (Also I, like @goessner, would prefer to avoid implementation terms like iteration.)

Reply:
> I don't see how the behaviour is well defined if we leave this out. It seems to me that the spec is silent on this matter.

A single item is just a list of length one.  That's the basis of this PR.  Since how a list of *pickers works is defined, a single *picker is inherently defined.

> I checked the text of the filter *picker and it doesn't specify the result order. It talks about iterating, but doesn't tie down the order of iteration.

Okay.  I prefer to add it there rather than in the selector because not all *pickers iterate, only the filter one does.

> Also I, like @goessner, would prefer to avoid implementation terms like iteration.

We need some description of going over each item in turn.  How else would you say this aside from "iterate"?

Reply:
> > I don't see how the behaviour is well defined if we leave this out. It seems to me that the spec is silent on this matter.
> 
> A single item is just a list of length one. That's the basis of this PR. Since how a list of *pickers works is defined, a single *picker is inherently defined.

Not so, I'm afraid. Please could you point me to the text which says how a selector with a single *picker operates.

> 
> > I checked the text of the filter *picker and it doesn't specify the result order. It talks about iterating, but doesn't tie down the order of iteration.
> 
> Okay. I prefer to add it there rather than in the selector because not all *pickers iterate, only the filter one does.
> 
> > Also I, like @goessner, would prefer to avoid implementation terms like iteration.
> 
> We need some description of going over each item in turn. How else would you say this aside from "iterate"?

We can use phrases like "the *picker matches certain values in the array". The spec only needs to say what the result is, not how the result is arrived at.